atd-vze/src/queries/crashes_changes.js

```diff
@@ -0,0 +1,283 @@
+            shldr_width_right
+            shldr_use_right_id
+            median_type_id
+            median_width
+            rural_urban_type_id
+            func_sys_id
+            adt_curnt_amt
+            adt_curnt_year
+            adt_adj_curnt_amt
+            pct_single_trk_adt
+            pct_combo_trk_adt
+            trk_aadt_pct
+            curve_type_id
+            curve_lngth
+            cd_degr
+            delta_left_right_id
+            dd_degr
+            feature_crossed
+            structure_number
+            i_r_min_vert_clear
+            approach_width
+            bridge_median_id
+            bridge_loading_type_id
+            bridge_loading_in_1000_lbs
+            bridge_srvc_type_on_id
+            bridge_srvc_type_under_id
+            culvert_type_id
+            roadway_width
+            deck_width
+            bridge_dir_of_traffic_id
+            bridge_rte_struct_func_id
+            bridge_ir_struct_func_id
+            crossingnumber
+            rrco
+            poscrossing_id
+            wdcode_id
+            standstop
+            yield
+            sus_serious_injry_cnt
+            nonincap_injry_cnt
+            poss_injry_cnt
+            non_injry_cnt
+            unkn_injry_cnt
+            tot_injry_cnt
+            death_cnt
+            mpo_id
+            investigat_service_id
+            investigat_da_id
+            investigator_narrative
+            cr3_stored_flag
+        }
+        atd_txdot_changes(where: {record_type: {_eq: "crash"}, record_id: {_eq: $crashId}}) {
+            change_id
+            record_id
+            created_timestamp
+            record_type
+            status {
+                description
+            }
+            status_id
+            record_json
+        }
+    }
+`;
+
+/**
+ * Allows us to search all secondary records that are not a crash,
+ * namely: all involved primary persons, persons, and units.
+ * @type {gql}
+ */
+export const GET_CRASH_SECONDARY_RECORDS = gql`
+    query findSecondaryRecords($crashId: Int) {
+        atd_txdot_changes(
+            where: {
+                record_id: {_eq: $crashId}
+            }
+        ) {
+            change_id
+            record_id
+            record_type
+            record_json
+        }
+    }
+`;
+
+/**
+ * This is a GraphQL template that removes all change records in the
+ * changes table by CrashID upon being discarded.
+ * @type {string}
+ */
+export const CRASH_MUTATION_DISCARD = `
+    mutation discardChanges($crashId: Int) {
+      delete_atd_txdot_changes(where: {record_id: {_eq: $crashId}}) {
+        affected_rows
+      }
+    }
+`
+
+/**
+ * This is a GraphQL query template that updates existing records
+ * of a specific type. E.g., it expects one or many Units, or Persons
+ * or Crashes. The function name matches a Hasura insertion method.
+ * @type {string}
+ */
+
+export const RECORD_MUTATION_UPDATE = `
+        %FUNCTION_NAME%(         
```

Comment:
I don't fully understand this syntax. Can you show me an example or docs on how this works? Thank you in advance.

Reply:
This is a template, anything within the percentage symbols is a variable. For example `%FUNCTION_NAME%` is a variable that will later be changed. This is is not proper python, I should have used Templates instead.